doc/classes/AStar3D.xml

```diff
@@ -6,29 +6,40 @@
 	<description>
 		A* (A star) is a computer algorithm used in pathfinding and graph traversal, the process of plotting short paths among vertices (points), passing through a given set of edges (segments). It enjoys widespread use due to its performance and accuracy. Godot's A* implementation uses points in 3D space and Euclidean distances by default.
 		You must add points manually with [method add_point] and create segments manually with [method connect_points]. Once done, you can test if there is a path between two points with the [method are_points_connected] function, get a path containing indices by [method get_id_path], or one containing actual coordinates with [method get_point_path].
-		It is also possible to use non-Euclidean distances. To do so, create a class that extends [AStar3D] and override methods [method _compute_cost] and [method _estimate_cost]. Both take two indices and return a length, as is shown in the following example.
+		It is also possible to use non-Euclidean distances. To do so, create a class that extends [AStar3D] and override methods [method _compute_cost] and [method _estimate_cost]. Both take two IDs and return a length, as is shown in the following example which uses Manhattan distance instead of Euclidean distance.
 		[codeblocks]
 		[gdscript]
-		class MyAStar:
-		    extends AStar3D
+		class_name MyAStar3D
+		extends AStar3D
 
-		    func _compute_cost(u, v):
-		        return abs(u - v)
+		func _compute_cost(u, v):
+		    var u_pos = get_point_position(u)
+		    var v_pos = get_point_position(v)
+		    return abs(u_pos.x - v_pos.x) + abs(u_pos.y - v_pos.y) + abs(u_pos.z - v_pos.z)
```

Comment:
I don't think that is valid code :). You're taking the absolute value of the difference of two vectors and returning it from a function that returns a float.

If you meant to suggest `return (u_pos - v_pos).length()` or similar, then that would be calculating the *Euclidean* distance. It's a fine example of using the API but it should also be the default implementation of A*, and is not an example of overriding these functions for Non-Euclidean purposes. My example finds [Manhattan/Taxicab distance](https://en.wikipedia.org/wiki/Taxicab_geometry), in which the distance is the *sum* of the difference of each cartesian coordinate.

I can see the reasoning for using Euclidean distance as the example, would just need to change some of the previous description.

Reply:
In that case this could be shortened to:
```suggestion
		    var absolute = (u_pos - v_pos).abs()
		    return absolute.x + absolute.y + absolute.z
```
But no, it somehow feels more complex than it actually is. My concern is mainly about the length of a single line.

Reply:
Line length is even more of a problem in the C# example because C# names are more verbose. Despite that, I think these make more sense as single lines

Reply:
Needs a damn good name for the variable, or just `a` or `vec` as last resort.

Reply:
I think the C# example is reasonably idiomatic for the docs already:
- The `Mathf` namespace is usually inline, most examples don't do `using Mathf;`.
- `toPosition` and `fromPosition` seem idiomatic. `toPos` and `fromPos` might be fine too, but the docs seem to prefer spelling out the whole name.

I guess I could split into the two lines like your suggestion just for the sake of making the line length shorter. IDK, the line already seems as concise as it can be given how verbose C# is. I don't see an obvious improvement here.

Reply:
I shortened `toPosition` and `fromPosition` to `toPoint` and `fromPoint`.

Reply:
Note that you changed the names only for C#

Reply:
I think it's okay in this case. We are getting the values from `get_point_position()`/`GetPointPosition()`. So either "point" or "position" are reasonable names. `u_pos` is an idiomatic GDScript name (short for `u_position`), and `toPoint` is an idiomatic C# name. Note that the parameters for `_compute_cost()` are `u` and `v` for GDScript but `fromID` and `toID` for C#, so there is already some difference in naming built into Godot already.

I could change `u_pos` to `u_point` but I don't think that serves anyone, it's longer for no reason. And I'd like to avoid `toPos` and `fromPos` because the current C# examples in other classes tend to spell out the full name, and it's more instructive to use the full name.